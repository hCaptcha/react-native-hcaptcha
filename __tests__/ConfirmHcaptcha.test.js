import React from 'react';
import renderer from 'react-test-renderer';
import ConfirmHcaptcha from '../index';

describe('ConfirmHcaptcha snapshot tests', () => {
  it('renders ConfirmHcaptcha with minimum props', () => {
    const component = renderer.create(
      <ConfirmHcaptcha
        siteKey="00000000-0000-0000-0000-000000000000"
        baseUrl="https://hcaptcha.com"
        languageCode="en"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders ConfirmHcaptcha with all props', () => {
    const component = renderer.create(
      <ConfirmHcaptcha
        size="compact"
        siteKey="00000000-0000-0000-0000-000000000000"
        baseUrl="https://hcaptcha.com"
        languageCode="en"
        showLoading={false}
        loadingIndicatorColor="#999999"
        backgroundColor="rgba(0.1, 0.1, 0.1, 0.4)"
        theme="light"
        rqdata='{"some": "data"}'
        sentry={true}
        jsSrc="https://all.props/api-endpoint"
        endpoint="https://all.props/endpoint"
        reportapi="https://all.props/reportapi"
        assethost="https://all.props/assethost"
        imghost="https://all.props/imghost"
        host="all-props-host"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
