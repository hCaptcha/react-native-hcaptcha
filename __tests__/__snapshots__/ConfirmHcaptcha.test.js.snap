// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmHcaptcha snapshot tests renders ConfirmHcaptcha with all props 1`] = `
<Modal
  animationIn="fadeIn"
  animationOut="fadeOut"
  coverScreen={true}
  deviceHeight={1334}
  deviceWidth={750}
  hasBackdrop={true}
  hideModalContentWhileAnimating={true}
  isVisible={false}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  style={
    [
      {
        "display": "none",
        "margin": 0,
      },
      {
        "display": undefined,
      },
    ]
  }
  useNativeDriver={true}
>
  <RCTSafeAreaView
    style={
      [
        {
          "flex": 1,
          "justifyContent": "center",
          "overflow": "hidden",
        },
        {
          "backgroundColor": "rgba(0.1, 0.1, 0.1, 0.4)",
        },
      ]
    }
  >
    <WebView
      automaticallyAdjustContentInsets={true}
      injectedJavaScript="(function () {
  var originalPostMessage = window.ReactNativeWebView.postMessage;
  var patchedPostMessage = function patchedPostMessage(message, targetOrigin, transfer) {
    originalPostMessage(message, targetOrigin, transfer);
  };
  patchedPostMessage.toString = function () {
    return String(Object.hasOwnProperty).replace('hasOwnProperty', 'postMessage');
  };
  window.ReactNativeWebView.postMessage = patchedPostMessage;
})();"
      javaScriptEnabled={true}
      mixedContentMode="always"
      onShouldStartLoadWithRequest={[Function]}
      originWhitelist={
        [
          "*",
        ]
      }
      renderLoading={[Function]}
      source={
        {
          "baseUrl": "https://hcaptcha.com",
          "html": "<!DOCTYPE html>
      <html>
      <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta http-equiv="X-UA-Compatible" content="ie=edge">
        <script type="text/javascript">
          Object.entries({"rnver_0_0_0":true,"dep_mocked-md5":true,"sdk_1_7_2":true}).forEach(function (entry) { window[entry[0]] = entry[1] })
        </script>
        <script src="https://all.props/api-endpoint?render=explicit&onload=onloadCallback&host=all-props-host&hl=en&sentry=true&endpoint=https%3A%2F%2Fall.props%2Fendpoint&assethost=https%3A%2F%2Fall.props%2Fassethost&imghost=https%3A%2F%2Fall.props%2Fimghost&reportapi=https%3A%2F%2Fall.props%2Freportapi" async defer></script>
        <script type="text/javascript">
          var onloadCallback = function() {
            try {
              console.log("challenge onload starting");
              hcaptcha.render("submit", getRenderConfig("00000000-0000-0000-0000-000000000000", "light", "compact"));
              // have loaded by this point; render is sync.
              console.log("challenge render complete");
            } catch (e) {
              console.log("challenge failed to render");
              window.ReactNativeWebView.postMessage("error");
            }
            try {
              console.log("showing challenge");
              hcaptcha.execute(getExecuteOpts());
            } catch (e) {
              console.log("failed to show challenge");
              window.ReactNativeWebView.postMessage("error");
            }
          };
          var onDataCallback = function(response) {
            window.ReactNativeWebView.postMessage(response);
          };
          var onCancel = function() {
            window.ReactNativeWebView.postMessage("cancel");
          };
          var onOpen = function() {
            window.ReactNativeWebView.postMessage("open");
            console.log("challenge opened");
          };
          var onDataExpiredCallback = function(error) { window.ReactNativeWebView.postMessage("expired"); };
          var onChalExpiredCallback = function(error) { window.ReactNativeWebView.postMessage("cancel"); };
          var onDataErrorCallback = function(error) {
            console.log("challenge error callback fired");
            window.ReactNativeWebView.postMessage("error");
          };
          const getRenderConfig = function(siteKey, theme, size) {
            var config = {
              sitekey: siteKey,
              size: size,
              callback: onDataCallback,
              "close-callback": onCancel,
              "open-callback": onOpen,
              "expired-callback": onDataExpiredCallback,
              "chalexpired-callback": onChalExpiredCallback,
              "error-callback": onDataErrorCallback
            };
            if (theme) {
              config.theme = theme;
            }
            return config;
          };
          const getExecuteOpts = function() {
            var opts;
            const rqdata = "{"some": "data"}";
            if (rqdata) {
              opts = {"rqdata": rqdata};
            }
            return opts;
          };
        </script>
      </head>
      <body style="background-color: rgba(0.1, 0.1, 0.1, 0.4);">
        <div id="submit"></div>
      </body>
      </html>",
        }
      }
      startInLoadingState={false}
      style={
        [
          {
            "backgroundColor": "transparent",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </RCTSafeAreaView>
</Modal>
`;

exports[`ConfirmHcaptcha snapshot tests renders ConfirmHcaptcha with minimum props 1`] = `
<Modal
  animationIn="fadeIn"
  animationOut="fadeOut"
  coverScreen={true}
  deviceHeight={1334}
  deviceWidth={750}
  hasBackdrop={true}
  hideModalContentWhileAnimating={true}
  isVisible={false}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  style={
    [
      {
        "display": "none",
        "margin": 0,
      },
      {
        "display": undefined,
      },
    ]
  }
  useNativeDriver={true}
>
  <RCTSafeAreaView
    style={
      [
        {
          "flex": 1,
          "justifyContent": "center",
          "overflow": "hidden",
        },
        {
          "backgroundColor": "rgba(0, 0, 0, 0.3)",
        },
      ]
    }
  >
    <WebView
      automaticallyAdjustContentInsets={true}
      injectedJavaScript="(function () {
  var originalPostMessage = window.ReactNativeWebView.postMessage;
  var patchedPostMessage = function patchedPostMessage(message, targetOrigin, transfer) {
    originalPostMessage(message, targetOrigin, transfer);
  };
  patchedPostMessage.toString = function () {
    return String(Object.hasOwnProperty).replace('hasOwnProperty', 'postMessage');
  };
  window.ReactNativeWebView.postMessage = patchedPostMessage;
})();"
      javaScriptEnabled={true}
      mixedContentMode="always"
      onShouldStartLoadWithRequest={[Function]}
      originWhitelist={
        [
          "*",
        ]
      }
      renderLoading={[Function]}
      source={
        {
          "baseUrl": "https://hcaptcha.com",
          "html": "<!DOCTYPE html>
      <html>
      <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta http-equiv="X-UA-Compatible" content="ie=edge">
        <script type="text/javascript">
          Object.entries(["test_key"]).forEach(function (entry) { window[entry[0]] = entry[1] })
        </script>
        <script src="https://js.hcaptcha.com/1/api.js?render=explicit&onload=onloadCallback&host=00000000-0000-0000-0000-000000000000.react-native.hcaptcha.com&hl=en" async defer></script>
        <script type="text/javascript">
          var onloadCallback = function() {
            try {
              console.log("challenge onload starting");
              hcaptcha.render("submit", getRenderConfig("00000000-0000-0000-0000-000000000000", "light", "invisible"));
              // have loaded by this point; render is sync.
              console.log("challenge render complete");
            } catch (e) {
              console.log("challenge failed to render");
              window.ReactNativeWebView.postMessage("error");
            }
            try {
              console.log("showing challenge");
              hcaptcha.execute(getExecuteOpts());
            } catch (e) {
              console.log("failed to show challenge");
              window.ReactNativeWebView.postMessage("error");
            }
          };
          var onDataCallback = function(response) {
            window.ReactNativeWebView.postMessage(response);
          };
          var onCancel = function() {
            window.ReactNativeWebView.postMessage("cancel");
          };
          var onOpen = function() {
            window.ReactNativeWebView.postMessage("open");
            console.log("challenge opened");
          };
          var onDataExpiredCallback = function(error) { window.ReactNativeWebView.postMessage("expired"); };
          var onChalExpiredCallback = function(error) { window.ReactNativeWebView.postMessage("cancel"); };
          var onDataErrorCallback = function(error) {
            console.log("challenge error callback fired");
            window.ReactNativeWebView.postMessage("error");
          };
          const getRenderConfig = function(siteKey, theme, size) {
            var config = {
              sitekey: siteKey,
              size: size,
              callback: onDataCallback,
              "close-callback": onCancel,
              "open-callback": onOpen,
              "expired-callback": onDataExpiredCallback,
              "chalexpired-callback": onChalExpiredCallback,
              "error-callback": onDataErrorCallback
            };
            if (theme) {
              config.theme = theme;
            }
            return config;
          };
          const getExecuteOpts = function() {
            var opts;
            const rqdata = null;
            if (rqdata) {
              opts = {"rqdata": rqdata};
            }
            return opts;
          };
        </script>
      </head>
      <body style="background-color: rgba(0, 0, 0, 0.3);">
        <div id="submit"></div>
      </body>
      </html>",
        }
      }
      startInLoadingState={false}
      style={
        [
          {
            "backgroundColor": "transparent",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </RCTSafeAreaView>
</Modal>
`;
